bigframes/core/blocks.py

```diff
@@ -2011,6 +2012,61 @@ def _is_monotonic(
         self._stats_cache[column_name].update({op_name: result})
         return result
 
+    def _get_rows_as_json_values(self) -> Block:
+        sql, index_column_ids, _ = self.to_sql_query(include_index=True)
+
+        # all column names, types and values
+        all_column_names = index_column_ids + [col for col in self.column_labels]
+        column_names_csv = ", ".join([f'"{col}"' for col in all_column_names])
+        column_references_csv = ", ".join(
+            [f"CAST({col} AS STRING)" for col in all_column_names]
+        )
+        all_column_types = list(self.index.dtypes) + list(self.dtypes)
+        column_types_csv = ", ".join([f'"{col}"' for col in all_column_types])
+
+        # index column names
+        index_column_names_csv = ", ".join([f'"{col}"' for col in index_column_ids])
+
+        row_json_column_name = guid.generate_guid()
+        select_columns = index_column_ids + [row_json_column_name]
+        select_columns_csv = ", ".join(select_columns)
+        json_sql = f"""\
+With T0 AS (
+{textwrap.indent(sql, "    ")}
+),
+T1 AS (
+    SELECT *,
+           JSON_OBJECT(
+               "names", [{column_names_csv}],
+               "types", [{column_types_csv}],
+               "values", [{column_references_csv}],
+               "index", [{index_column_names_csv}]
+           ) AS {row_json_column_name} FROM T0
+)
+SELECT {select_columns_csv} FROM T1
+"""
+        ibis_table = self.session.ibis_client.sql(json_sql)
+        order_for_ibis_table = ordering.ExpressionOrdering(
+            ordering_value_columns=tuple(
+                [ordering.ascending_over(column_id) for column_id in index_column_ids]
```

Comment:
This actually changes the ordering of the table, which we do not want. The table isn't necessarily ordered by index

Reply:
I'm looking into how to find the existing ordering information, do you have a tip on that? My current thought is to scan `block.expr.node.child_nodes` for a `OrderByNode`

Reply:
You should consider the tree structure to be an implementation detail. I think the best example of restoring ordering information would be the .cache() implementation, which materializes an ordering and creates a new ArrayValue using it.

Reply:
Ack. Used caching to honor the order, added a dedicated test `test_df_apply_axis_1_ordering`. PTAL.